simulator/hosts.go

```diff
@@ -0,0 +1,159 @@
+package simulator
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/ghodss/yaml"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+)
+
+type Host struct {
+	// Name of this host type.
+	Name string `json:"name"`
+	// parentName is the name of the parent type.
+	ParentName string `json:"parent"`
+	// parent is a pointer to the parent (nil if no parent) used for query
+	// result inheritance.
+	parent *Host
+	//QueryResults maps from regexp pattern to query results that should be
+	//returned.
+	Queries []matcher `json:"queries"`
+
+	// The following members facilitate logging unmatched queries.
+	logger           log.Logger
+	unmatchedMutex   sync.Mutex
+	unmatchedQueries map[string]bool
+}
+
+type matcher struct {
+	// Pattern is a regexp for the query patterns this should match.
+	Pattern regexp.Regexp `json:"pattern"`
+	// Results is the results to return for matched queries
+	Results []map[string]string `json:"results"`
+}
+
+type hostYAML struct {
```

Comment:
Perhaps we can rename this to something like `hostSimulationSpecification` or something more related to the content of the file?

Reply:
hostSpec sounds good to me.